### ISSUE TITLE
S1172: False positive for event handler and local function

### ISSUE THREAD

**Issue by Danghor** [SonarSource/sonar-dotnet#3585]
### Description

A false S1172 warning is shown when combining local functions with event handlers and using the supplied parameter only in the local function.

### Repro steps

```csharp
using System;
using System.IO;

namespace S1172
{
    public class Class1
    {
        public void DoSomething()
        {
            Something(new StreamReader(""));
        }

        // The following line triggers the warning: S1172 Remove this parameter 'disposable', whose value is ignored in the method.
        private void Something(IDisposable disposable)
        {
            void Console_CancelKeyPress(object sender, ConsoleCancelEventArgs e)
            {
                disposable.Dispose();
            }

            Console.CancelKeyPress += Console_CancelKeyPress;
        }
    }
}
```

### Expected behavior

The warning should not be shown because the parameter `disposable` is used in the local function which is used in the event handler. It is therefore used in the code and the code would not compile if the parameter was removed.

### Actual behavior

A S1172 warning is shown.

### Known workarounds

Suppress this particular warning.

### Related information

* C#/VB.NET Plugins Version: Sonar Lint for Visual Studio 2019 4.25.0.20544
* Visual Studio Version: Professional 2019
* MSBuild / dotnet version: .NET Standard 2.0
* If running through the Scanner for MSBuild, its version
* Operating System: Windows 10 x64

**Comment by pavel-mikula-sonarsource** [comment]
Hi @Danghor ,

Thank you for reporting this case. I can confirm it as False Positive. We already have similar case reported with the same root cause in #3134 so I'll close this issue and you can follow that one.

**Comment by Danghor** [comment]
Hi @pavel-mikula-sonarsource ,

sorry, I didn't see that one. Apparently, the GitHub search finds the issue with `is:issue is:open s1172` but does not with `is:issue is:open 1172`. I need to improve my search skills 😉
